.github/workflows/prbot.yml

```diff
@@ -16,6 +16,13 @@ jobs:
           node-version-file: '.nvmrc'
       - run: npm ci
       - run: npm run pr:report
+  lockfile-version:
+    name: 'Check lockfile version'
+    runs-on: 'ubuntu-latest'
+    steps:
+      - uses: actions/checkout@v3
+      - uses: ./.github/actions/setup
+      - run: node scripts/check-lockfile-versions.js $(find ./ -xdev -wholename '**/package-lock.json')
```

Comment:
could we not have this in lint-staged also / instead ?

Reply:
On a related note is there any way to detect if the pkg lock has been deleted and regenerated (inadvertently during a merge conflict e.g.)

Reply:
wouldn't having an engine in pkg json and engine strict setting prevent users from using the wrong node/npm version?

Reply:
we already have node-glob as a dep. Can we use that instead of find?

Reply:
> could we not have this in lint-staged also / instead
Sure.
- [ ] Move/add gate on lint-staged

> On a related note is there any way to detect if the pkg lock has been deleted and regenerated (inadvertently during a merge conflict e.g.)

Mhh I don't see how that'd be possible. We could do more checks on the lockfile, such as checking if some package are "where they should be", but I reckon the cost of coding and maintaining the rules ain't worth the additional gain.

> wouldn't having an engine in pkg json and engine strict setting prevent users from using the wrong node/npm version?

We could enforce `npm9` indeed, it would help it would not prevent users from comitting wrong lockfileversion.
One could've set something like lockfile-version=2 in its global npmrc, and thus cause the issue to persist even with the latest npm version

>we already have node-glob as a dep. Can we use that instead of find?

Sure, this makes sense if we put the gate on the pre-commit

Reply:
- [ ]  add lockfile-version=3 in npmrc

Reply:
Sounds good 👍

Reply:
> FYI just came across https://www.npmjs.com/package/eslint-plugin-package-lock which seems to do the same check. Would be better if we can use it as it would be a eslint plugin and would go with the lint workflow. What do you think?

simple as pie, this is nice. reworking with this.